### PR TITLE
Score of tweets is nil

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

**Comment by davgur** [comment]
[Score of Tweets is nil when sorting is not by relevance](https://trello.com/c/Qdu8645K/8-score-of-tweets-is-nil-when-sorting-is-not-by-relevance)

File: search/tweets_engine.go [hunk 1]
```diff
@@ -67,15 +77,21 @@ func (e *ESEngine) CombineResultsToSingleHit(resultsByLang map[string]*elastic.S
...
-
-			if hit.Score == nil {
-				return nil, errors.Errorf("hit score is nil for hit: %s", hit.Uid)
-			}
-
 			if *hit.Score > maxScore {
 				maxScore = *hit.Score
 			}
+			var ed es.EffectiveDate
+			if err := json.Unmarshal(*hit.Source, &ed); err != nil {
```
Comment (YuriHechter) [inline search/tweets_engine.go]: Consider serialization to es.Result. The Source may contain more fields that we may need.

File: search/tweets_engine.go [hunk 2]
```diff
@@ -45,6 +47,14 @@ func (e *ESEngine) SearchTweets(query Query, sortBy string, from int, size int,
 			err := errors.New(fmt.Sprintf("Failed tweets multi get: %+v", currentResults.Error))
 			return nil, err
 		}
+
+		for _, h := range currentResults.Hits.Hits {
```
[the remark this answers is not in the document]
Comment (davgur) [reply search/tweets_engine.go]: [commit](https://github.com/Bnei-Baruch/archive-backend/pull/168/commits/33b40306be3e3bac41f6cbf0c431290cd20226ed)

File: search/tweets_engine.go [hunk 3]
```diff
@@ -67,15 +77,21 @@ func (e *ESEngine) CombineResultsToSingleHit(resultsByLang map[string]*elastic.S
 	for _, result := range resultsByLang {
 
 		var maxScore float64
+		maxEd := es.EffectiveDate{EffectiveDate: &utils.Date{time.Now()}}
+		source, _ := json.Marshal(maxEd)
```
[the remark this answers is not in the document]
Comment (davgur) [reply search/tweets_engine.go]: [commit](https://github.com/Bnei-Baruch/archive-backend/pull/168/commits/b841003d27113ad2d4bc69508902d022a120a841)

**Pull request closed without merging**